src/oop/object_interactions/medium/q3/MainTest.java

```diff
@@ -0,0 +1,158 @@
+        assertTrue(testClass.hasMethod(getMyStoryMethodName, null, String.class, new String[]{"public"}),
+                String.format("You have modified the provided %s method method. Please revert it to the original state.", getMyStoryMethodName));
+
+    }
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of(2020, 5, 17, "It's a new day!"),
+                Arguments.of(2017, 4, 3, "A new brand of car is out!"),
+                Arguments.of(2023, 4, 5, "The classes are cancelled!")
+        );
+    }
+
+    @Test
+    public void newspaperClassHasRequiredConstructor() {
+        Class<?>[] classArguments = {int.class, int.class, int.class, String.class};
+        assertTrue(testClass.hasConstructor(classArguments),
+                "Your " + className + " constructor does not have the correct parameters.");
+        assertTrue(testClass.hasModifier(classArguments, "public"),
+                "Your " + className + " constructor does not have the correct modifier.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void newspaperClassHasCorrectConstructor(int year, int month, int day, String story) throws Throwable {
+        Object[][] arguments = {
+                {year, int.class},
+                {month, int.class},
+                {day, int.class},
+                {story, String.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        _assertEquals(year, testClass.getFieldValue(classInstance, myYearAttributeName),
+                "Your " + className + " constructor does not correctly initialize the " + myYearAttributeName + " attribute.");
+        _assertEquals(month, testClass.getFieldValue(classInstance, myMonthAttributeName),
+                "Your " + className + " constructor does not correctly initialize the " + myMonthAttributeName + " attribute.");
+        _assertEquals(day, testClass.getFieldValue(classInstance, myDayAttributeName),
+                "Your " + className + " constructor does not correctly initialize the " + myDayAttributeName + " attribute.");
+        _assertEquals(story, testClass.getFieldValue(classInstance, myStoryAttributeName),
+                "Your " + className + " constructor does not correctly initialize the " + myStoryAttributeName + " attribute.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void correctHearSecretMethod(int year, int month, int day, String story) throws Throwable {
+        Object classInstance = testClass.createInstance();
+        Object[][] methodArguments = {
+                {new Newspaper(year, month, day, story), Newspaper.class}
+        };
+        Object output = testClass.callMethod(makeCopyMethodName, methodArguments, new String[]{"public"}, classInstance);
```

Comment:
this should get split up into separate asserts for name, arguments, and visibility